db/event_manager_test.go

```diff
@@ -831,3 +831,69 @@ func assertChannelLengthWithTimeout(t *testing.T, c chan interface{}, expectedLe
 		}
 	}
 }
+
+func mockDBStateChangeEvent(dbName string, state string, reason string, adminInterface string) *DBStateChangeEvent {
+	body := make(Body, 5)
+	body["dbname"] = dbName
+	body["admininterface"] = adminInterface
+	body["state"] = state
+	body["reason"] = reason
+	body["localtime"] = time.Now().Format(base.ISO8601Format)
+	event := &DBStateChangeEvent{Doc: body}
+	return event
+}
+
+type UnsupportedEvent struct {
+	AsyncEvent
+}
+
+func (event *UnsupportedEvent) String() string {
+	return fmt.Sprintf("Couchbase Sync Gateway doesn't support this kind of events!")
+}
+
+func (event *UnsupportedEvent) EventType() EventType {
+	return EventType(255)
+}
+
+// Simulate the scenario for handling unsupported events.
+func TestWebhookHandleUnsupportedEvenType(t *testing.T) {
+	prev, err := strconv.Atoi(base.StatsResourceUtilization().Get(base.StatKeyWarnCount).String())
+	assert.NoError(t, err, "No error while getting the count of warnings")
+	wh := &Webhook{url: "https://127.0.0.1:8086/webhook?"}
+	event := &UnsupportedEvent{}
+	wh.HandleEvent(event)
```

Comment:
As discussed, let's update HandleEvent to return bool, and use that instead of asserting warning counts.

Reply:
This has been fixed and changes are available in the latest revision.